### PR TITLE
added an url filter option in scenes

### LINKED ISSUE #1059: [Feature] add a url filter option
**Is your feature request related to a problem? Please describe.**
- There's no way (that I know of) to search/filter by the content of a scene's url.
- Use case example: I want to determine which urls contain a specific domain so I can tag them accordingly.

**Describe the solution you'd like**
- add a `url` option to the filter

**Additional context**
`url` filtering options coulde be:
- includes
- excludes
- equals
- not equals

### PR DESCRIPTION
**Pull request by julien0221**
I added an url filter option into scenes 
Resolves #1059

Do we want to apply the same filter for galleries, movies, performers and/or studios?

### CODE REVIEW

File: ui/v2.5/src/models/list-filter/criteria/utils.ts [hunk 1]
```diff
@@ -113,5 +113,17 @@ export function makeCriteria(type: CriterionType = "none") {
...
+    case "url": {
+      const ret = new StringCriterion(type, type);
+      ret.modifierOptions = [
+        Criterion.getModifierOption(CriterionModifier.Equals),
+        Criterion.getModifierOption(CriterionModifier.NotEquals),
+        Criterion.getModifierOption(CriterionModifier.Includes),
+        Criterion.getModifierOption(CriterionModifier.Excludes),
+        Criterion.getModifierOption(CriterionModifier.MatchesRegex),
+        Criterion.getModifierOption(CriterionModifier.NotMatchesRegex),
+      ];
```
Comment (WithoutPants) [inline ui/v2.5/src/models/list-filter/criteria/utils.ts:125]: Why is this necessary? The standard modifier options (which should include `IS [NOT] NULL`) should be sufficient.

**Review by WithoutPants — COMMENTED** [review]
Thanks for the contribution. Yes, please extend this for the other objects that have urls.

File: ui/v2.5/src/models/list-filter/criteria/utils.ts (hunk 1 above)
Comment (julien0221) [reply ui/v2.5/src/models/list-filter/criteria/utils.ts]: yes you are right, I removed it and use the standard modifier.

**Comment by julien0221** [comment]
Ok I applied the same url filter to movies, galleries, performers and studios. And I used the standard modifier.

**Review by WithoutPants — APPROVED** [review]
